examples/plot_connectivity.py

```diff
@@ -30,73 +32,103 @@
+weight, delay, lamtha = 1.0, 1.0, 70
+src = 'L5_pyramidal'
+for target in ['L5_pyramidal', 'L2_basket']:
+    net_all.add_connection(src, target, location, receptor,
+                           delay, weight, lamtha)
+
+# Basket cell connections
+location, receptor = 'soma', 'gabaa'
+weight, delay, lamtha = 1.0, 1.0, 70
+src = 'L2_basket'
+for target in ['L5_pyramidal', 'L2_basket']:
+    net_all.add_connection(src, target, location, receptor,
+                           delay, weight, lamtha)
+
+dpl_all = simulate_dipole(net_all, n_trials=1)
+net_all.cell_response.plot_spikes_raster()
 
 ###############################################################################
-# That's a lot of spiking! Since basket cells are inhibitory, removing these
-# connections increases network wide excitability. We can additionally add
-# new connections using ``net.add_connection()``. Let's try connecting a
-# single layer 2 basket cell, to every layer 2 pyramidal cell. We can utilize
-# ``net.gid_ranges`` to help find the gids of interest.
-# :meth:`hnn_core.Network.add_connection` allows connections to be specified
-# with either cell names, or the gids directly. If multiple gids are provided
-# for either the sources or the targets, they will be connected in an
-# all-to-all pattern.
-print(net.gid_ranges)
-src_gid = net.gid_ranges['L2_basket'][0]
-target_gids = 'L2_pyramidal'
+# With the previous connection pattern there appears to be synchronous rhythmic
+# firing of the L5 pyramidal cells with a period of 10 ms. The synchronous
+# activity is visible as vertical lines where several cells fire simultaneously
+# We can additionally use the ``probability``. argument to create a sparse
+# connectivity pattern instead of all-to-all. Let's try creating the same
+# network with a 10% chance of cells connecting to each other. The resulting
+# connectivity pattern can also be visualized with
+# ``net.connectivity[idx].plot()``
+probability = 0.1
+net_sparse = default_network(params, add_drives_from_params=True)
+net_sparse.clear_connectivity()
+
+# Pyramidal cell connections
+location, receptor = 'distal', 'ampa'
+weight, delay, lamtha = 1.0, 1.0, 70
+src = 'L5_pyramidal'
+for target in ['L5_pyramidal', 'L2_basket']:
+    net_sparse.add_connection(src, target, location, receptor,
+                              delay, weight, lamtha, probability)
+
+# Basket cell connections
 location, receptor = 'soma', 'gabaa'
 weight, delay, lamtha = 1.0, 1.0, 70
-net.add_connection(src_gid, target_gids, location, receptor,
-                   delay, weight, lamtha)
+src = 'L2_basket'
+for target in ['L5_pyramidal', 'L2_basket']:
+    net_sparse.add_connection(src, target, location, receptor,
+                              delay, weight, lamtha, probability)
 
-net_add = net.copy()
-dpl_add = simulate_dipole(net_add, n_trials=1)
-net_add.cell_response.plot_spikes_raster()
+dpl_sparse = simulate_dipole(net_sparse, n_trials=1)
+net_sparse.cell_response.plot_spikes_raster()
```

Comment:
could you add a tiny text line after this so that the connectivity plots are forced on the next line?

Reply:
I actually meant a line of text. But that's fine, we can fix it later